python/ray/remote_function.py

```diff
@@ -92,7 +93,7 @@ def __init__(self, language, function, function_descriptor, num_cpus,
         # Override task.remote's signature and docstring
         @wraps(function)
         def _remote_proxy(*args, **kwargs):
-            return self._remote(args=args, kwargs=kwargs)
+            return self._remote(args=args, kwargs=kwargs, internal_called=True)
 
         self.remote = _remote_proxy
```

Comment:
@edoakes If we remove _remote completely, then this wrapper here that used to keep function's signature also needs to be removed. for detail see https://github.com/ray-project/ray/pull/4985